pkg/controller/mongodb/mongodb_controller.go

```diff
@@ -540,15 +540,17 @@ func mongodbContainer(version string, volumeMounts []corev1.VolumeMount) contain
 	mongoDbCommand := []string{
 		"/bin/sh",
 		"-c",
-		// we execute the pre-stop hook once the mongod has been gracefully shut down by the agent.
-		`while [ ! -f /data/automation-mongod.conf ]; do sleep 3 ; done ; sleep 2 ;
-# start mongod with this configuration
-mongod -f /data/automation-mongod.conf ;
+		`
+# run post-start hook to handle version changes.
+# during a version change, mongod will be stopped and the
+# container will be restarted, at which points this hook runs.
```

Comment:
I think we should stick to 1 line comment in the startup script and we have the details in the docs

Reply:
Agreed. Updated the comment and `architecture.md`.